de/scratchwork.ipynb

```diff
@@ -0,0 +1,417 @@
+    "data_combined = data_combined.drop([\"WT\"], axis=1)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data_combined.columns"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data_combined.loc[\"FOSL1\",\"FOSL1\"]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import matplotlib.pyplot as plt\n",
+    "from sklearn.preprocessing import normalize\n",
+    "from sklearn.decomposition import PCA\n",
+    "import numpy as np\n",
+    "%matplotlib inline\n",
+    "\n",
+    "#------------------------- Perform PCA using sklearn\n",
+    "def pca(data,num_components):\n",
+    "    \"Function takes in parameter for number of components. Returns list containing: [PCA object, fitted model]\"\n",
+    "    pca = PCA(n_components=num_components)\n",
+    "    X_r = pca.fit(normalize(data)).transform(data)\n",
+    "    return [pca, X_r]\n",
+    "\n",
+    "#-------------------------- Calculate cumulative variance based on number of PCs included and create r2x plot\n",
+    "def r2x(num_components, pca):\n",
+    "    \"\"\"Function creates r2x plot comparing number of components to % variance captured.\n",
+    "    Parameters include: number of components, pca object from pca()\"\"\"\n",
+    "    total_variance = np.array([])\n",
+    "    tot = 0.0\n",
+    "    for i in range(0,num_components):\n",
+    "        tot += pca.explained_variance_ratio_[i]\n",
+    "        total_variance = np.append(total_variance, tot)\n",
+    "\n",
+    "    fig = plt.figure(figsize = (8,8))\n",
+    "    ax = plt.subplot(111) \n",
+    "    ax.set_xlabel(\"Number of PCs\", fontsize = 15)\n",
+    "    ax.set_ylabel(\"% Variance\", fontsize = 15)\n",
+    "    plt.xticks(np.arange(num_components+1))\n",
+    "    plt.plot(list(range(1,num_components+1)),total_variance)\n",
+    "    ax.grid()\n",
+    "    plt.title(\"r2x Plot\", fontsize = 18)\n",
+    "    "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pca_list = pca(data.T,10)\n",
+    "r2x(10, pca_list[0])\n",
+    "X_r = pca_list[1]\n",
+    "KO_genes = data.columns\n",
+    "KO_genes_unique = list(set(KO_genes))\n",
+    "df = pd.DataFrame(X_r)\n",
+    "df[\"KO Gene\"] = KO_genes"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import matplotlib.colors as colors\n",
+    "import matplotlib.cm as cmx\n",
+    "\n",
+    "#-------------------------- Create PC plots\n",
+    "\n",
+    "# Set the color map to match the number of species\n",
+    "z = range(1,len(KO_genes_unique))\n",
+    "rainbow = plt.get_cmap('rainbow')\n",
+    "cNorm  = colors.Normalize(vmin=0, vmax=len(KO_genes_unique))\n",
```

Comment:
If you move this file out of the `de` folder, you can import functions from the `de` package this way: `from de.file import function`. This will prevent the code from being duplicated here.